src/app/pages/products/ProductCard.js

```diff
@@ -0,0 +1,69 @@
+import { Card, Table } from '@newfold/ui-component-library';
+import { productApiFetch, isJarvis } from '../../util/helpers';
+
+const formatDate = ( dateString ) => {
+	const date = new Date( dateString );
+	const month = ( date.getMonth() + 1 ).toString().padStart( 2, '0' );
```

Comment:
We need to validate the passed data is a valid string and before parsing the `date` we need to validate that the `date` is valid to prevent type errors.